sklearn/ensemble/gradient_boosting.py

```diff
@@ -511,7 +512,7 @@ def _update_terminal_region(self, tree, terminal_regions, leaf, X, y,
         numerator = np.sum(sample_weight * residual)
         denominator = np.sum(sample_weight * (y - residual) * (1 - y + residual))
 
-        if denominator == 0.0:
+        if isclose(denominator, 0., rtol=0., atol=np.float64(1e-150)):
```

Comment:
silly question but is that a scalar? I feel the code ``denominator < np.float64(1e-150)`` easier to understand.